libparsec/crates/client/src/invite/greeter.rs

```diff
@@ -265,88 +398,150 @@ struct BaseGreetInitialCtx {
+            let claimer_step = run_greeter_step_until_ready(
+                &self.cmds,
+                greeting_attempt,
+                invite_greeter_step::GreeterStep::Number1GetHashedNonce,
+                &self.device.time_provider,
+            )
+            .await?;
+            let result: Result<_, GreetInProgressError> = match claimer_step {
+                invite_greeter_step::ClaimerStep::Number1SendHashedNonce { hashed_nonce } => {
+                    Ok(hashed_nonce)
                 }
-            }?
+                _ => Err(anyhow::anyhow!("Unexpected claimer step: {:?}", claimer_step).into()),
+            };
+            result?
         };
 
-        let claimer_nonce = {
-            use authenticated_cmds::latest::invite_2b_greeter_send_nonce::{Rep, Req};
-
-            let rep = self
-                .cmds
-                .send(Req {
+        {
+            let claimer_step = run_greeter_step_until_ready(
+                &self.cmds,
+                greeting_attempt,
+                invite_greeter_step::GreeterStep::Number2SendNonce {
                     greeter_nonce: greeter_nonce.clone(),
-                    token: self.token,
-                })
-                .await?;
+                },
+                &self.device.time_provider,
+            )
+            .await?;
+            let result: Result<_, GreetInProgressError> = match claimer_step {
+                invite_greeter_step::ClaimerStep::Number2GetNonce => Ok(()),
+                _ => Err(anyhow::anyhow!("Unexpected claimer step: {:?}", claimer_step).into()),
+            };
+            result?
+        };
 
-            match rep {
-                Rep::Ok { claimer_nonce } => Ok(claimer_nonce),
-                Rep::InvitationDeleted => Err(GreetInProgressError::AlreadyDeleted),
-                Rep::EnrollmentWrongState => Err(GreetInProgressError::PeerReset),
-                Rep::InvitationNotFound => Err(GreetInProgressError::NotFound),
-                bad_rep @ Rep::UnknownStatus { .. } => {
-                    Err(anyhow::anyhow!("Unexpected server response: {:?}", bad_rep).into())
+        let claimer_nonce = {
+            let claimer_step = run_greeter_step_until_ready(
+                &self.cmds,
+                greeting_attempt,
+                invite_greeter_step::GreeterStep::Number3GetNonce,
+                &self.device.time_provider,
+            )
+            .await?;
+            let result: Result<_, GreetInProgressError> = match claimer_step {
+                invite_greeter_step::ClaimerStep::Number3SendNonce { claimer_nonce } => {
+                    Ok(claimer_nonce)
                 }
-            }?
+                _ => Err(anyhow::anyhow!("Unexpected claimer step: {:?}", claimer_step).into()),
+            };
+            result?
         };
 
         if HashDigest::from_data(&claimer_nonce) != claimer_hashed_nonce {
+            if cancel_greeting_attempt(
```

Comment:
common about dropping the error because this is best effort

Reply:
e5dc2bed6a1b059894901854e3df8e15d765e1b0